sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/implementation/VirtualMachineMockTests.java

```diff
@@ -0,0 +1,195 @@
+import com.azure.resourcemanager.compute.models.StorageAccountTypes;
+import com.azure.resourcemanager.compute.models.StorageProfile;
+import com.azure.resourcemanager.compute.models.VirtualMachine;
+import com.azure.resourcemanager.compute.models.VirtualMachineSizeTypes;
+import com.azure.resourcemanager.resources.fluentcore.arm.ResourceUtils;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.common.FileSource;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.extension.Parameters;
+import com.github.tomakehurst.wiremock.extension.ResponseTransformer;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.http.Response;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class VirtualMachineMockTests {
+    private static final String NEXT_LINK_PATH = "/nextLink";
+    // query contains unescaped space
+    private static final String QUERY = "'virtualMachineScaleSet/id' eq 'id'";
+    private static final SerializerAdapter SERIALIZER = SerializerFactory.createDefaultManagementSerializerAdapter();
+    private final StateHolder stateHolder = new StateHolder();
+
+    @Test
+    public void listByVmssByIdWithNextLinkEncoded() {
+        WireMockServer mockServer = startMockServer();
+
+        try {
+            ComputeManager computeManager = mockComputeManager(mockServer);
+
+            PagedIterable<VirtualMachine> virtualMachines = computeManager.virtualMachines().listByVirtualMachineScaleSetId("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/javacsmrg97796/providers/Microsoft.Compute/virtualMachineScaleSets/vmss035803b7");
+            // 1 element per page, 2 pages in total
+            Assertions.assertEquals(2, virtualMachines.stream().count());
+            Assertions.assertTrue(stateHolder.firstPageRequested);
+            Assertions.assertTrue(stateHolder.secondPageRequested);
+        } finally {
+            if (mockServer.isRunning()) {
+                mockServer.shutdown();
+            }
+        }
+    }
+
+    private ComputeManager mockComputeManager(WireMockServer mockServer) {
+        Map<String, String> environment = new HashMap<>();
+        environment.put("resourceManagerEndpointUrl", mockServer.baseUrl());
+        // just to mitigate npe, no actual use here
+        environment.put("microsoftGraphResourceId", mockServer.baseUrl());
+        AzureProfile mockProfile = new AzureProfile(UUID.randomUUID().toString(), UUID.randomUUID().toString(), new AzureEnvironment(environment));
+        ComputeManager computeManager = ComputeManager.authenticate(new HttpPipelineBuilder().build(), mockProfile);
```

Comment:
Did we do a real auth? Should we put a mock credential? https://github.com/Azure/azure-sdk-for-java/blob/7df467719475fae92cb68d84772e8aa9bdd954b0/sdk/dnsresolver/azure-resourcemanager-dnsresolver/src/test/java/com/azure/resourcemanager/dnsresolver/generated/DnsResolversCreateOrUpdateTests.java#L62

Reply:
I used an empty pipeline without `AuthenticationPolicy` here, so no token is needed. 

If use default pipeline, `BearerTokenAuthenticationPolicy` would check if the request http protocol is `https`:
https://github.com/Azure/azure-sdk-for-java/blob/d2bfd01a7ca51d79356a1bf0ba10b57c2077e383/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/BearerTokenAuthenticationPolicy.java#L125-L128
Since I'm doing local mock server with `http`, I disable it.

Reply:
Sure. That's fine.